### DISCUSSION TITLE
MSB Solution

### DISCUSSION THREAD

**Discussion by cbarrygrcs** [snwau/picoCTF-2023-Writeup#6] [General]
Hello snwau
I attempted to follow your posted solution and I am unable to. I end up with a section of a book, and the last part of the text appears to be encoded. I tried to search "picoCTF" as you stated but nothing is found. This leads me to believe I need to decode the ending section to find the flag. Can you offer any more details?

**Comment by snwau** [comment]
Hey cbarrygrcs,

No further decoding was needed after the MSB extraction via sigBits.py.

I used my standard text editor (Sublime Text) to do text search and found the flag. It's located in the top half of the file, so no need to try and decode the stuff at the end of the file.

Alternatively if you have a hex editor (ImHex is great) I just did an ASCII string search in there on the output and the flag was located starting at file offset 0x3CD6E.

Can be a bit hard to view, given the file has no line endings and it's just one huge string of text.

Cheers,

**Reply by cbarrygrcs** [reply]
Thanks for the quick response. I'll give that a try too.

**Reply by snwau** [reply]
> Thanks for the quick response. I'll give that a try too.

Also another way from the command line is to get grep to restrict its output to the search string plus additional characters after to display the key:

$ cat outputSB.txt | grep -o -E "picoCTF.{0,50}"
